python/pydantic_core/core_schema.py

```diff
@@ -659,6 +660,72 @@ def float_schema(
     )
 
 
+class DecimalSchema(TypedDict, total=False):
+    type: Required[Literal['decimal']]
+    gt: Union[int, Decimal]
+    ge: Union[int, Decimal]
+    lt: Union[int, Decimal]
+    le: Union[int, Decimal]
+    max_digits: int
+    decimal_places: int
+    multiple_of: Union[int, Decimal]
+    allow_inf_nan: bool  # whether 'NaN', '+inf', '-inf' should be forbidden. default: False
+    check_digits: bool  # FIXME document. default: False
```

Comment:
not sure what this is referring to, I guess I'll find out once it's documented

Reply:
`check_digits` actually shouldn't be part of the schema, it's an internal state of the validator, will remove from here.